tests/test_searchspace.py

```diff
@@ -361,3 +361,149 @@ def test_polars_pandas_equivalence(parameters):
+
+    # Test batch size of 1 as well as one small and one large batch size
+    for batch_size in [1, 2, 42]:
+        bounds = subspace.comp_rep_bounds.values
+        samples = subspace._sample_from_polytope(batch_size, bounds)
+
+        constraint_result = (
+            samples["Conti_finite1"].sum() + 2 * samples["Conti_finite2"].sum()
+        )
+        assert np.isclose(constraint_result, 0.3, atol=1e-6)
+
+
+@pytest.mark.parametrize("parameter_names", [["Conti_finite1", "Conti_finite2"]])
+@pytest.mark.parametrize("constraint_names", [["InterConstraint_4"]])
+def test_sample_from_polytope_with_interpoint_constraints_inequality(
+    parameters, constraints
+):
+    """Test _sample_from_polytope method with interpoint inequality constraints."""
+    subspace = SubspaceContinuous(
+        parameters=parameters,
+        constraints_lin_ineq=constraints,
+    )
+
+    assert subspace.has_interpoint_constraints
+
+    # Test batch size of 1 as well as one small and one large batch size
+    for batch_size in [1, 2, 42]:
+        bounds = subspace.comp_rep_bounds.values
+        samples = subspace._sample_from_polytope(batch_size, bounds)
+
+        constraint_result = (
+            2 * samples["Conti_finite1"].sum() - samples["Conti_finite2"].sum()
+        )
+        assert constraint_result >= 0.3 - 1e-6
+
+
+def test_sample_from_polytope_mixed_constraints_with_interpoint():
+    """Test _sample_from_polytope method with regular and interpoint constraints."""
+    # NOTE: This test does not use our fixtures as those seem to create an infeasible
+    # space
+    parameters = [
+        NumericalContinuousParameter("Conti_finite1", (0, 1)),
+        NumericalContinuousParameter("Conti_finite2", (-1, 0)),
+    ]
+
+    regular_constraint = ContinuousLinearConstraint(
+        parameters=["Conti_finite2"],
+        operator=">=",
+        coefficients=[1.0],
+        rhs=-0.8,
+    )
+    interpoint_constraint = ContinuousLinearConstraint(
+        parameters=["Conti_finite1"],
+        operator="=",
+        coefficients=[1],
+        rhs=0.6,
+        interpoint=True,
+    )
+
+    subspace = SubspaceContinuous(
+        parameters=parameters,
+        constraints_lin_ineq=[regular_constraint],
+        constraints_lin_eq=[interpoint_constraint],
+    )
+
+    assert subspace.has_interpoint_constraints
+
+    # Test batch size of 1 as well as one small and one large batch size
+    for batch_size in [1, 2, 42]:
+        bounds = subspace.comp_rep_bounds.values
+        samples = subspace._sample_from_polytope(batch_size, bounds)
+
+        # Verify regular constraint is satisfied for each row
+        assert (samples["Conti_finite2"] >= -0.8 - 1e-6).all()
+
+        # Verify interpoint constraint is satisfied across the batch
+        interpoint_constraint_result = samples["Conti_finite1"].sum()
+        assert np.isclose(interpoint_constraint_result, 0.6, atol=1e-6)
+
+
+def test_sample_from_polytope_interpoint_ordering_robustness():
```

Comment:
Have actually checked if the test conducts the expected check, i.e. that it fails before adjusting the corresponding code and succeeds afterwards? I'm asking because I think the test is actually useless / doesn't do what you think it does. You see, the you are reordering parameters, but the first thing that happens when you pass them to the searchspace is that the corresponding converter sorts them back. So everything downstream of this point is trivially identical!?